Sandbox/src/ExampleLayer.cpp

```diff
@@ -0,0 +1,198 @@
+			out vec4 v_Color;
+
+			void main()
+			{
+				v_Position = a_Position;
+				v_Color = a_Color;
+				gl_Position = u_ViewProjection * u_Transform * vec4(a_Position, 1.0);	
+			}
+		)";
+
+	std::string fragmentSrc = R"(
+			#version 330 core
+			
+			layout(location = 0) out vec4 color;
+
+			in vec3 v_Position;
+			in vec4 v_Color;
+
+			void main()
+			{
+				color = vec4(v_Position * 0.5 + 0.5, 1.0);
+				color = v_Color;
+			}
+		)";
+
+	m_Shader = Hazel::Shader::Create("VertexPosColor", vertexSrc, fragmentSrc);
+
+	std::string flatColorShaderVertexSrc = R"(
+			#version 330 core
+			
+			layout(location = 0) in vec3 a_Position;
+
+			uniform mat4 u_ViewProjection;
+			uniform mat4 u_Transform;
+
+			out vec3 v_Position;
+
+			void main()
+			{
+				v_Position = a_Position;
+				gl_Position = u_ViewProjection * u_Transform * vec4(a_Position, 1.0);	
+			}
+		)";
+
+	std::string flatColorShaderFragmentSrc = R"(
+			#version 330 core
+			
+			layout(location = 0) out vec4 color;
+
+			in vec3 v_Position;
+			
+			uniform vec3 u_Color;
+
+			void main()
+			{
+				color = vec4(u_Color, 1.0);
+			}
+		)";
+
+	m_FlatColorShader = Hazel::Shader::Create("FlatColor", flatColorShaderVertexSrc, flatColorShaderFragmentSrc);
+
+	auto textureShader = m_ShaderLibrary.Load("assets/shaders/Texture.glsl");
+
+	m_Texture = Hazel::Texture2D::Create("assets/textures/Checkerboard.png");
+	m_ChernoLogoTexture = Hazel::Texture2D::Create("assets/textures/ChernoLogo.png");
+
+	textureShader->Bind();
+	textureShader->SetInt("u_Texture", 0);
+}
+
+void ExampleLayer::OnAttach()
+{
+}
+
+void ExampleLayer::OnDetach()
+{
+}
+
+void ExampleLayer::OnUpdate(Hazel::Timestep ts) 
+{
+	// Update
+	m_CameraController.OnUpdate(ts);
+
+	// Render
+	Hazel::RenderCommand::SetClearColor({ 0.1f, 0.1f, 0.1f, 1 });
+	Hazel::RenderCommand::Clear();
+
+	Hazel::Renderer::BeginScene(m_CameraController.GetCamera());
+
+	glm::mat4 scale = glm::scale(glm::mat4(1.0f), glm::vec3(0.1f));
+
+	m_FlatColorShader->Bind();
+	m_FlatColorShader->SetFloat3("u_Color", m_SquareColor);
+
+	for (int y = 0; y < 20; y++)
+	{
+		for (int x = 0; x < 20; x++)
+		{
+			glm::vec3 pos(x * 0.11f, y * 0.11f, 0.0f);
+			glm::mat4 transform = glm::translate(glm::mat4(1.0f), pos) * scale;
+			Hazel::Renderer::Submit(m_FlatColorShader, m_SquareVA, transform);
+		}
+	}
+
+	auto textureShader = m_ShaderLibrary.Get("Texture");
+
+	m_Texture->Bind();
+	Hazel::Renderer::Submit(textureShader, m_SquareVA, glm::scale(glm::mat4(1.0f), glm::vec3(1.5f)));
+	m_ChernoLogoTexture->Bind();
+	Hazel::Renderer::Submit(textureShader, m_SquareVA, glm::scale(glm::mat4(1.0f), glm::vec3(1.5f)));
+
+	// Triangle
+	// Hazel::Renderer::Submit(m_Shader, m_VertexArray);
+
+	Hazel::Renderer::EndScene();
+}
+
+void ExampleLayer::OnImGuiRender() 
+{
+	ImGui::Begin("Settings");
+	ImGui::ColorEdit3("Square Color", glm::value_ptr(m_SquareColor));
+	ImGui::End();
+}
+
+void ExampleLayer::OnEvent(Hazel::Event& e) 
+{
+	m_CameraController.OnEvent(e);
+}
+
```

Comment:
You should remove these extra spaces.

Reply:
Thanks, for pointing that out @CleverSource. The extra spaces have been removed.

Reply:
Resolved in https://github.com/TheCherno/Hazel/pull/170/commits/ddeb98cc54ade9ca965f144823cf28f496283475